### DISCUSSION TITLE
Client Only Components Data Handling

### DISCUSSION THREAD

**Discussion by charlieforward9** [vikejs/vike#737] [Discussion]
I am developing a map dashboard similar to Google Maps. As a junior developer, I'd like to know the smartest way to integrate the Map with the map controls (search bar, detail bar, etc...) when wrapped in a Client Only Component (CO).

My initial thought was to wrap only the Map in a CO and pass relevant properties **into** it as it is modified. However, now I am considering the alternative putting everything that communicates with the Map into a CO and passing the data **within** that.

I tried to see how Google Maps does it, but it is hard to tell... if I had to guess, it looks like they pass data **within** the client rendered component.

If passing properties **into** is the smartest way of doing this. I would like to propose for the docs to be updated for optional props to be passed into the [`ClientOnly` component](https://vite-plugin-ssr.com/dynamic-import) once it is loaded. If passing the properties **within** is, then ill handle that design myself.

As always, thank you!

**Comment by brillout** [comment]
Can't say without knowing specifics, but I'd be inclined to do this: get the initial data on the server-side with `onBeforeRender()` then pass that to the client-only component, then fetch subsequent data within the client-only component.

**Reply by charlieforward9** [reply]
I'll test it out, thanks!
